### PR TITLE
Allow arbitrary arguments to be passed to pip through pip.installed

### PR DESCRIPTION
**Pull request by Akm0d**
### What does this PR do?
Makes it possible to pass arbitrary arguments to pip from pip.installed

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/24751

### Previous Behavior
N/A

### New Behavior
 pip keyword and positional arguments not yet implemented in salt can be passed to pip through pip.installed

    .. code-block:: yaml

        pandas:
          pip.installed:
            - name: pandas
            - pip_future:
              - --latest-pip-kwarg:
                - param1
                - param2
              - --latest-pip-arg

    Will be translated into the following pip command:

    .. code-block:: bash

        pip install pandas --latest-pip-kwarg param1 --latest-pip-kwarg parm2 --latest-pip-arg

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

**Comment by waynew** [comment]
Uh... I think you might want to rebase your fixes :trollface: (or retarget your PR)

**Comment by twangboy** [comment]
@Akmod Looks like you have some lint

**Comment by twangboy** [comment]
https://jenkinsci.saltstack.com/job/pr-lint/job/PR-52327/1/warnings52Result/new/

**Comment by Akm0d** [comment]
I don't understand why it's not passing the py3-centos-7 test.  I ran the tests manually using python3.6.1 on my own centos7 server and it all passed just fine

**Comment by Akm0d** [comment]
Failing on Centos7 with python3 and windows, but not because of the commit.  Centos7 and Windows are having issues in Jenkins

**Comment by waynew** [comment]
I'm not sure if the remaining problem is legit something with these tests - it is a problem in the pip state tests, so probably worth looking into 👍

**Review by waynew — APPROVED** [review]
👍

**Review by thatch45 — APPROVED** [review]

File: salt/modules/pip.py [hunk 1]
```diff
@@ -895,6 +915,24 @@ def install(pkgs=None,  # pylint: disable=R0912,R0913,R0914
     if trusted_host:
         cmd.extend(['--trusted-host', trusted_host])
 
+    if extra_args:
+        # These are arguments from the latest version of pip that
+        # have not yet been implemented in salt
```
Comment (twangboy) [inline salt/modules/pip.py]: `# These are any additional arguments to be passed as parameters to the pip command`

I don't think we need to specify that it's for arguments salt hasn't implemented yet

File: salt/modules/pip.py [hunk 2]
```diff
@@ -612,6 +614,24 @@ def install(pkgs=None,  # pylint: disable=R0912,R0913,R0914
     no_cache_dir
         Disable the cache.
 
+    extra_args
+        pip keyword and positional arguments not yet implemented in salt
```
Comment (twangboy) [inline salt/modules/pip.py]: `pip keyword and positional arguments to be passed as parameters to the pip command`

I don't think we need to denote `not yet implemented`

File: salt/states/pip_state.py [hunk 3]
```diff
@@ -602,6 +605,23 @@ def installed(name,
                 - reload_modules: True
                 - exists_action: i
 
+    extra_args
+        pip keyword and positional arguments not yet implemented in salt
```
Comment (twangboy) [inline salt/states/pip_state.py]: same here

**Review by twangboy — CHANGES_REQUESTED** [review]
A few minor things. Looks great.

**Comment by max-arnold** [comment]
Can someone backport this to master branch?

**Comment by Akm0d** [comment]
@max-arnold Done :+1:

**Pull request merged**